File: KthMissingPosNum.cpp
 int kthMissing(vector<int> &arr, int k) {
        // Your code goes here
        int missingCount = 0;
        int currentNum = 1;
        int idx = 0;
        int n = arr.size();
        while(missingCount < k){
            if(idx < n && arr[idx] == currentNum){
                ++idx;
            }
            else{
                ++missingCount;
                if(missingCount == k){
                    return  currentNum;
                }
            }
            ++currentNum;
        }
        return -1;
    }

File: KthMissingPosNum.java
 public int kthMissing(int[] arr, int k) {
        // code here
        int idx = 0;
        int curr = 1;
        int missCnt = 0;
        int n = arr.length;
        while(missCnt < k){
            if(idx < n && arr[idx] == curr){
                ++idx;
            }
            else{
                ++missCnt;
                if(missCnt == k){
                    return curr;
                }
            }
            ++curr;
        }
        return -1;
    }
